source/LocationFilter.cpp

```diff
@@ -469,7 +469,9 @@ const Planet *LocationFilter::PickPlanet(const System *origin, bool hasClearance
 		if(!planet.IsValid())
 			continue;
 		// Skip planets that do not offer special jobs or missions, unless they were explicitly listed as options.
-		if(planet.IsWormhole() || (requireSpaceport && !planet.HasSpaceport()) || (!hasClearance && !planet.CanLand()))
+		if(planet.IsWormhole()
+				|| (requireSpaceport && !(planet.GetPort().services & Port::ServicesType::OffersMissions))
```

Comment:
Why not use `planet.IsAvailable(Port::ServicesType::OffersMissions)`? (There might be a few other places where you can do this too.)